### ISSUE TITLE
Missing support for MXML Component node as direct child of Flex 3 Class

### ISSUE THREAD

**Issue by ashikase** [apache/royale-compiler#241]
Version tested: apache-royale-0.9.12

While working on an old code base that uses a mix of Flex 3 and Flex 4 classes, I found that `<mx:Component>` tags in the Flex 3 files were being detected as errors (specifically, `MXMLUnexpectedTagProblem`).

The compiler expects `Component` nodes at the class level to be wrapped in an `<fx:Declarations>` node (`MXMLDeclarationsNode`), as such:

```mxml
<mx:Box
  xmlns:fx="http://ns.adobe.com/mxml/2009"
  xmlns:mx="library://ns.adobe.com/flex/mx"
>
  <fx:Declarations>
    <fx:Component id="...">...</fx:Component>
  </fx:Declarations>
</mx:Box>
```

This is correct for Flex 4, but there is no `Declarations` tag in Flex 3; components can be written as direct children of the class:

```mxml
<mx:Box
  xmlns:mx="http://ns.adobe.com/mxml/2006"
>
  <mx:Component id="...">...</mx:Component>
</mx:Box>
```

I only started looking at the royale-compiler source today, so I am not very familiar with the codebase, but my naive fix for this was to add the following code:

```java
        else if (fileScope.isComponentTag(childTag) &&
                 (builder.getMXMLDialect().isEqualToOrBefore(MXMLDialect.MXML_2006)))
        {
            childNode = new MXMLComponentNode(this);
        }
```

... after [line 289](https://github.com/apache/royale-compiler/blob/658ba460b4f8362a379003c259aa2bb01307819b/compiler/src/main/java/org/apache/royale/compiler/internal/tree/mxml/MXMLClassDefinitionNode.java#L289) of the `MXMLClassDefinitionNode::parseChildTag` method.

**Comment by joshtynjala** [comment]
I confirm that the Flex SDK compiler allows `<mx:Component>` as a direct child of the root tag when using the MXML 2006 namespace. I also confirm that the Royale compiler reports an error for the same code.

Your fix looks correct to me. I'll get it committed.
